### PR TITLE
Kb product model

### PR DESCRIPTION
**Pull request by kenziebottoms**
# Description
- `createProduct`,
- `getProduct` by `id`
- `getAllProducts`
- `getProductsByCreator`
- `deleteProduct` by `id`
- `getAllStockedProducts`
- `getQuantityRemaining` by `id`

## Related Ticket(s)
#4, #7, #5

## Steps to Test Solution
1. `npm run db:reset`
1. `npm test`

## Testing
- [x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
- [x] I certify that all existing tests pass

### CODE REVIEW

File: test/productsM.test.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        assert.isObject(products[0]);
+      })
+      .catch(err => console.log('getProductsByCreator error', err));
+  });
+  it('creator 2 should have 6 products', () => {
+    getProductsByCreator(2)
+      .then(response => {
+        assert.equal(response.length, 6);
+      })
+      .catch(err => console.log('getProductsByCreator error', err));
```
Comment (DavidLarsKetch) [inline test/productsM.test.js:86]: This is console logging and passing the test, for some reason

File: test/productsM.test.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+describe('getQuantityRemaining()', () => {
+  it('should return a promise', () => {
+    assert.typeOf(getQuantityRemaining(3).catch(err => {}), 'promise');
+  });
+  it('should resolve into a number', () => {
+    getQuantityRemaining(3)
+      .then(qty => {
+        assert.isNumber(qty);
+      })
+      .catch(err => console.log('getQuantityRemaining error', err));
```
Comment (DavidLarsKetch) [inline test/productsM.test.js:137]: This is console logging and passing the test, for some reason

File: test/productsM.test.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  });
+  it('getQuantityRemaining(40) should return 7', () => {
+    getQuantityRemaining(40)
+      .then(qty => {
+        assert.equal(qty, 7);
+      })
+      .catch(err => console.log('getQuantityRemaining() error', err));
```
Comment (DavidLarsKetch) [inline test/productsM.test.js:144]: This is console logging and passing the test, for some reason

**Review by DavidLarsKetch — DISMISSED** [review]
:-1:

**Review by melissabell456 — APPROVED** [review]
✔️ 34 tests passing

**Review by jordan-castelloe — APPROVED** [review]
👍

**Pull request merged**